upgrade/v100to101/upgrade.go

```diff
@@ -0,0 +1,114 @@
+package v100to101
+
+import (
+	"github.com/pkg/errors"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta1"
+	lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
+	"github.com/longhorn/longhorn-manager/types"
+)
+
+// This upgrade is needed because we changed from using full image name (in v1.0.0)
+// to use the checksum of image name (in v1.0.1) for instance manager labels.
+// Therefore, we need to update all existing instance manager labels so that
+// the updated Longhorn manager can correctly find them.
+// Link to the original issue: https://github.com/longhorn/longhorn/issues/1323
+
+const (
+	upgradeLogPrefix = "upgrade from v1.0.0 to v1.0.1: "
+)
+
+func UpgradeInstanceManagerPods(namespace string, lhClient *lhclientset.Clientset, kubeClient *clientset.Clientset) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, upgradeLogPrefix+"upgrade instance manager pods failed")
+	}()
+
+	imList, err := lhClient.LonghornV1beta1().InstanceManagers(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return errors.Wrapf(err, upgradeLogPrefix+"failed to list all existing instance managers during the instance managers pods upgrade")
+	}
+
+	for _, im := range imList.Items {
+		if im.Spec.Image != "" {
```

Comment:
Consider reducing the nesting of this by inverting this condition
```
if im.Spec.Image == "" { 
     continue 
}
```

Reply:
Done